File: Data Structures & Algorithms/maximum-subarray/submission-6.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums: number[]): number {
        let maxSum: number = nums[0];
        let curSum: number = 0;
        for (const num of nums){
            curSum = Math.max(curSum, 0)
            curSum += num
            maxSum = Math.max(maxSum, curSum)
        }
        return maxSum
    }
}

File: Data Structures & Algorithms/is-anagram/submission-6.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        f1, f2 = {}, {}

        for c in s:
            if c not in f1:
                f1[c] = 1
                continue
            f1[c] += 1
        
        for c in t:
            if c not in f2:
                f2[c] = 1
                continue
            f2[c] += 1
        return f1 == f2